### PR TITLE
Document release process

### LINKED ISSUE #142: Document release process
**What would you like to be added**:
Documentation for the release process.

/kind documentation

### PR DESCRIPTION
**Pull request by ayberk**
**What type of PR is this?**
> /kind documentation

**What this PR does / why we need it**:
Mainly for adding the release process, but I also changed the structure of the docs little bit. Now we have tabs for "User Guide", "Development" and "Changelog".  Example: https://ayberk.github.io/cloud-provider-aws/

**Which issue(s) this PR fixes**:
Fixes #142

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: docs/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+## Verify the release on GitHub
+
+The new tag should trigger a new Github release. Verify that it has run by going to [Releases](https://github.com/kubernetes/cloud-provider-aws/releases). Then, click on the new version and verify all assets have been created:
+
+- Source code (zip)
+- Source code (tar.gz)
+
+## Merge the release commit to the main branch
+
+Once the images are promoted, send a PR to merge the release commit to the main branch.
```
Comment (nckturner) [inline docs/RELEASE.md:81]: To the main branch or the release branch? Or does main mean release in this context?

Comment (ayberk) [reply docs/RELEASE.md:81]: This is the last step to update the main branch so our README etc is updated with the last version. 

Btw, we're not promoting our images yet but I'll add steps for it as well when we do.

Comment (nckturner) [reply docs/RELEASE.md:81]: Ok if the commit is coming from the release branch, then maybe it should specify to cherry pick just the release commit?  Merging every release branch into the main/master branch won't work if they are diverged, right?

**Review by nckturner — APPROVED** [review]

**Pull request merged**